pkg/cmd/experimental/sync-groups/dataextractor_schema1.go

```diff
@@ -0,0 +1,72 @@
+package group
+
+import "github.com/go-ldap/ldap"
+
+// FirstClassGroupDataExtractor extracts the member list and group metainformation from an LDAP server
+// with first-class LDAP entries for groups
+// ----------FOR SCHEMA 1----------
+type FirstClassGroupDataExtractor struct {
+	// SpecificGroupQueryConfig holds the information necessary to make an LDAP query for a specific
+	// first-class group entry on the LDAP server
+	SpecificGroupQueryConfig LDAPSpecificQueryConfig
+	// SpecificUserQueryConfig holds the information necessary to make an LDAP query for a specific
+	// first-class user entry on the LDAP server, given a first-class group entry with member records
```

Comment:
This feels like the wrong structure.  The query should substitute in the user dn/uid regardless of source